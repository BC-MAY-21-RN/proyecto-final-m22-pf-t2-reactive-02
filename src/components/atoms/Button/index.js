import React from 'react';
import {Text, TouchableOpacity} from 'react-native';
import colors from '../../../const/colors';

export default function Button({text, onPress, style, disabled, children}) {
  return (
    <TouchableOpacity
      disabled={!disabled}
      style={{
        ...style.btn,
        ...(!disabled ? {backgroundColor: colors.disabledpink} : null),
      }}
      onPress={onPress}>
      {children}
      <Text style={style.textBtn}>{text}</Text>
    </TouchableOpacity>
  );
}
