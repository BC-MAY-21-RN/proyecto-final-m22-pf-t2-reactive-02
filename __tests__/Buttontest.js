import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../src/components/atoms/Button';

describe('Test from Button component', () => {
  it('Button snapshot test', () => {
    const renderButton = renderer
      .create(
        <Button
          children={null}
          disabled={false}
          onPress={() => {}}
          style={{}}
          text={''}
          key={1}
        />,
      )
      .toJSON();
    expect(renderButton).toMatchSnapshot();
  });
});
