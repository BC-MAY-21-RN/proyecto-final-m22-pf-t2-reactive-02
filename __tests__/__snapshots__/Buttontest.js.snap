// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test from Button component Button snapshot test 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#B09AAC",
      "opacity": 1,
    }
  }
>
  <Text />
</View>
`;
